lib/indexer/message_processor.rb

```diff
@@ -0,0 +1,53 @@
+require_relative "../../app"
+require_relative "exceptions"
+require_relative "change_notification_processor"
+
+# MessageProcessor
+#
+# This class is instantiated by the rake task and passed to the message queue
+# consumer.
+
+# The consumer is currently using the message queue as a notification
+# mechanism. When it gets a message that some content has changed, it simply
+# re-indexes a document, which (among other things) triggers a new lookup of
+# the links from the publishing-api.
+module Indexer
+  class MessageProcessor
+    def process(message)
+      with_logging(message) do
+        Indexer::ChangeNotificationProcessor.new(message.payload).trigger
```

Comment:
Any particular reason to create a new CNP for each message? Seems wasteful in allocations for no significant readability improvement. Admittedly this is a drop in the ocean compared to other areas of the code, and we haven't got perf tests yet to show any difference.